tensorflow/core/kernels/cuda_solvers.cc

```diff
@@ -202,6 +202,7 @@ Status CudaSolver::CopyLapackInfoToHostAsync(
 // numeric types.
 #define TF_CALL_LAPACK_TYPES(m) \
   m(float, S) m(double, D) m(std::complex<float>, C) m(std::complex<double>, Z)
+#define TF_CALL_LAPACK_TYPES_NO_COMPLEX(m) m(float, S) m(double, D)
```

Comment:
That's actually a bit tricky: The SVD of a complex matrix gives real singular values. The implementation of cuSolver also provides the singular values in a real vector. The tensorflow operation, however, requires the singular values to be complex values as well. So I would need to store the singular values in a temporal array and convert them to complex values for the output (I can use CastFunctor defined in cast_op.h for that).
What is the reason that Tensorflow requires the singular values to be stored in a complex tensor?